### PR TITLE
Create Rule21-9.md

### PR DESCRIPTION
**Pull request by xingcx**

### CODE REVIEW

File: docs/section21/Rule21-9.md [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+**Manual Check:** None  
+**Evaluation Context:** Building  
+**Data Lookup:** None  
+**Function Call:** None  
+
+**Applicability Checks:**  
+
+1. B-RMR is modeled with at least one air-side system that is Type-1, 5, 7, 11, or 12: `if PLACEHOLDER_AIRSIDE_SYSTEM_RULE-X-XX:`
+2. B-RMR is not modeled with purchased heating: `if NOT Rule-21-1:`
+
```
Comment (supriyagoel) [inline docs/section21/Rule21-9.md:25]: should address this applicability check for purchased heating, as commented on other RDS.
two issues
1. we will probably not be implementing the RDS for 21-1, so should have another way to verify this applicability check.
2. This statement also does not clarify that a building doesn't have purchased heating. For that we should probably have some standard schema checks for external fluid source

File: docs/section21/Rule21-9.md [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+## Rule Logic:  
+
+- For each fluid loop in B_RMR: `for fluid_loop_b in B_RMR.ASHRAE229.fluid_loops:`
+
+  - Check if fluid loop is heating type: `if fluid_loop_b.type == "HEATING":`
+
+    **Rule Assertion:**
+
+    - Case 1: For each heating hot water loop that is served by boiler(s), if total pump power per flow rate is equal to 19W/gpm: `if fluid_loop_b.pump_power_per_flow_rate == 19: PASS`
```
Comment (supriyagoel) [inline docs/section21/Rule21-9.md:34]: @xingcx a fluid loop can have multiple child loops. we need to go and sum the pump power of all child loops and verify that the total pump power off all child loops is =19 W/CFM

**Review by supriyagoel — CHANGES_REQUESTED** [review]

File: docs/section21/Rule21-9.md (hunk 2 above)
Comment (xingcx) [reply docs/section21/Rule21-9.md:34]: Updated.

File: docs/section21/Rule21-9.md (hunk 1 above)
Comment (xingcx) [reply docs/section21/Rule21-9.md:25]: Added Rule 21-1 as applicability check.

File: docs/section21/Rule21-9.md [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+2. B-RMR is not modeled with purchased heating: `if NOT Rule-21-1:`
+
+## Rule Logic:  
+
+- For each fluid loop in B_RMR: `for fluid_loop_b in B_RMR.ASHRAE229.fluid_loops:`
+
+  - Check if fluid loop is heating type: `if fluid_loop_b.type == "HEATING":`
+
+    - Get total pump power per flow rate on the loop and all child loops: `total_pump_power_per_flow_rate = fluid_loop_b.pump_power_per_flow_rate + sum(fluid_loop.pump_power_per_flow_rate for fluid_loop in fluid_loop_b.child_loops)`
+
```
[the remark this answers is not in the document]
Comment (xingcx) [reply docs/section21/Rule21-9.md:33]: Updated.

File: docs/section21/Rule21-9.md (hunk 2 above)
Comment (xingcx) [reply docs/section21/Rule21-9.md]: Added Rule 21-11 (Baseline shall have only one HHW loop) and 21-18 (Baseline shall have only one primary HHW loop and no secondary loop) as applicability check. So now we do not need to check child loop pump power for this rule.

File: docs/section21/Rule21-9.md [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+**Appendix G Section Reference:** Section G3.1.3.5 Building System-Specific Modeling Requirements for the Baseline model  
+
+**Applicability:** All required data elements exist for B_RMR  
+**Applicability Checks:**  
+
+1. B-RMR is modeled with at least one air-side system that is Type-1, 5, 7, 11, or 12
+2. B-RMR is not modeled with purchased heating.
+3. Pass Rule 21-11
+4. Pass Rule 21-18
+
```
Comment (mkarpman) [inline docs/section21/Rule21-9.md:17]: Let's discuss applicability diagram and update accordingly.

File: docs/section21/Rule21-9.md (hunk 1 above)
Comment (mkarpman) [reply docs/section21/Rule21-9.md:29]: See note above on applicability

Comment (mkarpman) [reply docs/section21/Rule21-9.md:23]: Same comment on 21-7 etc., that applicability should be `if Rule-21-1 == "UNDERTERMINED"

**Review by mkarpman — APPROVED** [review]

**Review by supriyagoel — APPROVED** [review]

File: docs/section21/Rule21-9.md (hunk 2 above)
Comment (xingcx) [reply docs/section21/Rule21-9.md]: The applicability check is updated. Right now the logic only checks pump power on heating loops that are connected to boiler(s).

**Review by supriyagoel — APPROVED** [review]

**Review by mkarpman — APPROVED** [review]

**Pull request merged**